web/client/epics/catalog.js

```diff
@@ -162,5 +172,19 @@ module.exports = (API) => ({
                             position: "tc"
                         }), showLayerMetadata({}, false));
                 });
+            }),
+        autoSearchEpic: (action$, {getState = () => {}} = {}) =>
+            action$.ofType(CHANGE_TEXT)
+            .debounce(() => {
+                const state = getState();
+                const delay = delayAutoSearchSelector(state);
+                return Rx.Observable.timer(delay);
+            })
+            .switchMap(({text}) => {
+                const state = getState();
+                const pageSize = pageSizeSelector(state);
+                const {type, url} = selectedCatalogSelector(state);
+                return Rx.Observable.of(textSearch(type, url, 1, pageSize, text))
+                    .startWith(setLoading(true));
```

Comment:
Is this needed? It looks like textSearch already dispatches setLoading(true)

Reply:
yes, on a second thought this can be removed